client/dashboard/app/breadcrumbs/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { Breadcrumbs as BreadcrumbsComponent } from '@automattic/components/src/breadcrumbs';
+import { useMatches } from '@tanstack/react-router';
+import type { BreadcrumbItemProps } from '@automattic/components/src/breadcrumbs/types';
+
+export default function Breadcrumbs() {
+	const matches = useMatches();
```

Comment:
I haven't checked the logic yet, but it would be better to have a hook instead of a component for anyone to use. In fact, in dashboard the Breadcrumbs will be consumed through PageHeader which will accept the breadcrumb items.

Reply:
Just to confirm my understanding, do you mean something like this?

```jsx
export default function SubscriptionGiftingSettings() {
	...
	const breadcrumbItems = useBreadcrumbItems();

	return (
		<PageLayout size="small">
			<PageHeader
				title={ __( 'Accept a gift subscription' ) }
				description={ __( '...' ) }
				breadcrumbItems={breadcrumbItems} // <--------------- THIS
			/>
			...
		</PageLayout>
	);
}

```

Reply:
If so, wouldn't this mean that EVERY sub-route of the settings need to make sure to pass the breadcrumbs to its header? In this PR, I only need to do in once; which is in the Settings root route.

Reply:
With the current approach (composition) in `PageLayout`, that would be the code, yes. For now it doesn't really matter though either way. If we have the logic we can convert that to a hook if we need to, and we might even introduce a prop in PageHeader something like `showBreadcrumbs` and bake it inside `PageHeader` if no `breadcrumbItems` items are provided.

Let's focus on the logic here, it's fine.